.travis.yml

```diff
@@ -14,53 +15,92 @@ env:
+      - python3-venv
+      - pandoc
+
+before_install: # activate virtual environment
+  - cd $BASEPATH
+  - python3 -m venv venv
+  - source ./venv/bin/activate
+
+install: # install dependencies
+  - python3 -m pip install scikit-build
+  - python3 -m pip install pytest==5.4.1
+  - python3 -m pip install pytest-cov
+  - python3 -m pip install scipy
+  - wget https://root.cern.ch/download/root_${ROOTBIN}.tar.gz
+  - tar xpvfz root_*.tar.gz > /dev/null 2>&1
+  - source root/bin/thisroot.sh
+
+before_script: # build the project
+  - cd $BASEPATH
+  - python3 setup.py  --generator "Unix Makefiles" --skip-generator-test install -- -- -j2
+
 jobs:
   include:
     - os: osx # test XCode build
       osx_image: xcode10.2
-      addons:
-        homebrew:
-          packages:
-            - boost
-            - python3
-            - pip3
       env:
         - TASK="clang, ROOT v6.18"
         - ROOTBIN="v6.18.04.macosx64-10.14-clang100"
       script:
-        - cd $BASEPATH
-        - python setup.py  --generator "Unix Makefiles" --skip-generator-test install
-          -- -- -j2
-        # The test fails on mac osx because of a wrong RPATH in the precompiled root binaries
+        - echo "The tests can't run on mac osx because of a wrong RPATH in the precompiled root binaries"
 
-    - os: linux # test gcc build (Linux) + deploy pages
+    - os: linux # test gcc build (Linux)
       dist: bionic
       compiler: gcc
       env:
-        - TASK="gcc7, ROOT v6.18 + deploy pages"
+        - TASK="gcc7, ROOT v6.18"
         - ROOTBIN="v6.18.04.Linux-ubuntu18-x86_64-gcc7.4"
-      addons:
-        apt:
-          update: true
-          packages:
-            - libboost-all-dev
-            - python3
-            - python3-pip
-            - python3-venv
-            - pandoc
       script:
-        - cd $BASEPATH
-        - python setup.py  --generator "Unix Makefiles" --skip-generator-test install
-          -- -- -j2
         - cd $BASEPATH/tests
         - python -m pytest
+
+    - os: linux # test and deploy documentation
+      dist: bionic
+      compiler: gcc
+      env:
+        - TASK="test and deploy documentation"
+        - ROOTBIN="v6.18.04.Linux-ubuntu18-x86_64-gcc7.4"
+      install:
+        - wget https://root.cern.ch/download/root_${ROOTBIN}.tar.gz
+        - tar xpvfz root_*.tar.gz > /dev/null 2>&1
+        - source root/bin/thisroot.sh
+        - cd $BASEPATH
+        - python3 -m pip install -r requirements.txt
+        - python3 -m pip install -r doc/requirements.txt
+        - python3 -m pip install virtualenvwrapper
+      before_script: echo "Running developer mode installation"
+      script: # developer mode build
+        - cd $BASEPATH
+        - source venv/bin/virtualenvwrapper.sh
+        - add2virtualenv $BASEPATH
+        - mkdir build
+        - cd build
+        - cmake .. -DUSE_GENEVA=OFF -DPYTHON_EXECUTABLE=$(which python)
+        - make -j2
+        - cd $BASEPATH/pycompwa
+        - ln -s ../build/ui*.so
```

Comment:
Do you want to test that the "developer" mode also works fine or why you do that here?
In my opinion this is not necessary to test, because its just a bit of venv juggling etc.

Reply:
Exactly. Yeah I thought that too at first, but (1) it's also `cmake` juggling (not trivial) and (2) there is no use in performing exactly the same building procedure as the other gcc job (which runs pytest after).

Reply:
But again. What do you test here? This test just verifies that that the juggling is correct. There are no different version of dependencies and the commands to build an test are underneath identical. In other words if this test alone fails the juggling is wrong, otherwise it will be in sync with the other ones.

 > (2) there is no use in performing exactly the same building procedure as the other gcc job (which runs pytest after)

Yeah, but thats why it was only one job before.

Reply:
Can we test another build procedure / compiler?

Reply:
i guess so

Reply:
Seems not https://docs.travis-ci.com/user/languages/cpp/#ci-environment-for-c-projects

Reply:
Okay, how to resolve this? If we merge the doc and pytest again, this PR can be closed

Reply:
I don't understand. You can choose a different compiler with apt.

If you ask me there is no need splitting up the docs building from the rest. But generally not a bad idea, to see the whats going wrong from the jobs and you already did it now. I would just do the installation in the simple and standard way instead of the dev mode (as explained above). Can you give one argument for using dev mode, what kind of test it serves? If so then I'm fine with it

Next time you should really open an issue first, before you make a PR and it just gets out of control. Then we can discuss it and create a rough idea how we want to achieve the goal first. At least for issues that are not straight forward.

Reply:
Working on the contribute tutorials, I found that in practice you will install `pycompwa` through dev mode anyway (i.e. CMake directly plus a venv wrapper). See motivation under https://github.com/ComPWA/pycompwa/issues/82.

I understand the technical reasoning here with regard to travis, but think the reasoning is too strict: we overlook the point that, in all likeliness, none of the users installs through the `setup.py` script (as happens in the travis jobs). So it seems wrong to me that we not test that build mode.

Reply:
> Next time you should really open an issue first, before you make a PR and it just gets out of control.

Could you make a note on this on the contribute page? The preference for issues over PRs is not clear.

Reply:
>Could you make a note on this on the contribute page? The preference for issues over PRs is not clear.

all right

Reply:
> I understand the technical reasoning here with regard to travis, but think the reasoning is too strict: we overlook the point that, in all likeliness, none of the users installs through the setup.py script (as happens in the travis jobs). So it seems wrong to me that we not test that build mode.

I feel like you don't understand what I'm trying to say.
python setup will call scikit build which will in turn call cmake and build. Installation is also done.
With the development mode you just call these commands by hand separately. And instead of installing files symlinked and added to the venv by path. So you test the virtualenvwrapper and symlinking. I just don't see any benefit of testing something like this. Providing these lines of how to add custom paths to your venv does not need ci testing in my view. Its fine if you mention this in the docs. Please correct me if I'm wrong or if I misunderstand your idea behind this test.

Reply:
https://github.com/ComPWA/pycompwa/blob/911ea165ac5b4c7efecafab7e49273abd5346aef/doc/source/conf.py#L71

I don't understand how this currently works with skbuild.